src/OSPSuite.Presentation/Presenters/Charts/ChartEditorPresenter.cs

```diff
@@ -405,15 +406,31 @@ private void onChartPropertiesChanged()
 
       private void onDataBrowserUsedChanged(UsedColumnsEventArgs e)
       {
-         using (_chartUpdater.UpdateTransaction(Chart))
+         doAsCurveComplementUpdate(() => updateColumnUsedProperty(e.Columns, e.Used), curveDataChanged:true);
+      }
+
+      private void updateColumnUsedProperty(IReadOnlyList<DataColumn> columns, bool used)
+      {
+         columns.Each(column =>
          {
-            e.Columns.Each(column =>
-            {
-               if (e.Used)
-                  AddCurveForColumn(column);
-               else
-                  Chart.RemoveCurvesForColumn(column);
-            });
+            if (used)
+               AddCurveForColumn(column);
+            else
+               Chart.RemoveCurvesForColumn(column);
+         });
+      }
+
+      /// <summary>
+      ///    Encapsulates a chart update where the curves that should be recalculated are the complement of the curves before and
+      ///    after the <paramref name="updateAction" />
+      /// </summary>
+      private void doAsCurveComplementUpdate(Action updateAction, bool curveDataChanged)
+      {
+         // We need ToList so that a new list is created in memory and not a reference to the existing Chart.Curves list
+         var existingCurves = Chart.Curves.ToList();
+         using (_chartUpdater.UpdateTransaction(Chart, curveDataChanged, x => x.Curves.Complement(existingCurves).ToList()))
```

Comment:
I Am not quite getting it. Isn't it then a setting of the chartUpdater to kmow to only update the one that are changing ? Feels like a state instead
updateMode: ALL, ADDED
curveDataChanged is always true except when remove so this can proably be inferred to. Not sure (maybe mode is REMOVED)

Reply:
I added states which trigger the ChartUpdater to calculate what has changed. Makes it easier to understand what's going on for sure.

We don't use the `Complement` anymore, but I left it since I wrote the tests first :)